cmake/CompileBoost.cmake

```diff
@@ -12,7 +12,11 @@ function(compile_boost)
   set(BOOTSTRAP_LIBRARIES "context,filesystem")
 
   set(BOOST_CXX_COMPILER "${CMAKE_CXX_COMPILER}")
-  if(CLANG)
+  # Can't build Boost with Intel compiler, use clang instead.
+  if(ICX)
```

Comment:
We have `ICC` defined from our past experiments. Can we rename/remove that and consolidate with the new `ICX`?